system/Router/Router.php

```diff
@@ -573,32 +578,45 @@ public function autoRoute(string $uri)
 	//--------------------------------------------------------------------
 
 	/**
-	 * Attempts to validate the URI request and determine the controller path.
+	 * Scans the controller directory, attempting to locate a controller matching the supplied uri $segments
 	 *
 	 * @param array $segments URI segments
 	 *
-	 * @return array URI segments
+	 * @return array returns an array of remaining uri segments that don't map onto a directory
 	 */
-	protected function validateRequest(array $segments): array
+	protected function scanControllers(array $segments): array
 	{
 		$segments = array_filter($segments, function ($segment) {
-			// @phpstan-ignore-next-line
-			return ! empty($segment) || ($segment !== '0' || $segment !== 0);
+			return $segment !== '';
 		});
+		// numerically reindex the array, removing gaps
 		$segments = array_values($segments);
 
-		$c                 = count($segments);
-		$directoryOverride = isset($this->directory);
+		// if a prior directory value has been set, just return segments and get out of here
+		if (isset($this->directory))
+		{
+			return $segments;
+		}
```

Comment:
Why is this necessary? As this is untested, can you add a test case covering this possibility.

Reply:
I'm not entirely sure and will dig deeper. I definitely think we could use some more test cases and will try to add them.